src/Microsoft.Performance.SDK/Processing/TableDescriptor.cs

```diff
@@ -139,8 +144,24 @@ public sealed class TableDescriptor
         ///     Gets a value indicating whether this instance describes a
         ///     metadata table.
         /// </summary>
+        /// <remarks>
+        ///     Metadata tables are automatically marked as internal (<see cref="IsInternalTable"/>).
+        /// </remarks>
         public bool IsMetadataTable { get; }
 
+        /// <summary>
+        ///     Gets a value indicating whether this instance describes an internal table.
+        /// </summary>
+        /// <remarks>
+        ///     Internal tables are not required to have a static build method.
```

Comment:
I don't understand what an internal table is then. Why does it not need a static build method?

Reply:
An internal table is entirely maintained by the processing source and data processor. They are responsible for initializing and building the table, and therefore they can build it however they please. An internal table can still take advantage of the benefits that an extension table provides - making sure that a table's dependencies are all available, and restricting the table to data from those dependencies.

I'm not sure that we need to keep the internal table feature, but I also want to be careful about removing it as a feature so close to SDK 1.0. And even if we did that, I'd want to do it in a separate PR.